adding api gateway stage resource (#700)
inspec/inspec-aws: docs/resources/aws_api_gateway_stage.md

---
title: About the aws_api_gateway_stage Resource
platform: aws
---

# aws\_api\_gateway\_stage

> Use the `aws_api_gateway_stage` InSpec audit resource to test properties of the singular resource of AWS ApiGateway Stage.

Reviewer: ```suggestion
Use the `aws_api_gateway_stage` InSpec audit resource to test properties of a single AWS API Gateway stage.
```

> The AWS::ApiGateway::Stage resource deploys an API Gateway RestApi resource to a stage so that clients can call the API over the internet. The stage acts as an environment.

Reviewer: ```suggestion
The `AWS::ApiGateway::Stage` resource deploys an API Gateway REST API resource to a stage so that clients can call the API over the internet. The stage acts as an environment.
```

## Syntax

Ensure that the stage exists.

    describe aws_api_gateway_stage(rest_api_id: 'REST_API_ID', stage_name: 'STAGE_NAME') do
      it { should exist }
    end

## Parameters

`rest_api_id` _(required)_

> The string identifier of the associated RestApi.

Reviewer: ```suggestion
The ID of the REST API resource that you're deploying with this stage.
```

`stage_name` _(required)_

> The identifier of the stage resource to get information about.

Reviewer: ```suggestion
The API Gateway stage name.
```

> For additional information, see the [AWS ApiGateway Stage documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html).

Reviewer: ```suggestion
For additional information, see the [AWS API Gateway stage documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html).
```

## Properties

| Property | Description|
| --- | --- |
| deployment_id | The identifier for the deployment attached to stage resource. |
| description | The description for the stage resource. |
| created_date | The date and time that the stage resource was created. |
| stage_name | The name for the stage resource. |
| client_certificate_id | The client certificate identifier for the stage resource. |
| cache_cluster_enabled | Specifies whether a cache cluster is enabled for the stage. |
| cache_cluster_size | The size of the cache cluster for the stage, if enabled. |
| cache_cluster_status | The status of the cache cluster for the stage, if enabled. |

> | method_settings | A map that defines the method settings for a Stage resource. |

Reviewer: ```suggestion
| method_settings | A map that defines the method settings for a stage resource. |
```

| method_settings (logging_level) | Specifies the logging level for this method, which affects the log entries pushed to Amazon CloudWatch Logs. |
| method_settings (data_trace_enabled) | Specifies whether data trace logging is enabled for this method, which affects the log entries pushed to Amazon CloudWatch Logs. |
| method_settings (throttling_burst_limit) | Specifies the throttling burst limit. |
| method_settings (throttling_rate_limit) | Specifies the throttling rate limit. |
| method_settings (caching_enabled) | Specifies whether responses should be cached and returned for requests. |
| method_settings (cache_ttl_in_seconds) | Specifies the time to live (TTL), in seconds, for cached responses. |
| method_settings (cache_data_encrypted) | Specifies whether the cached responses are encrypted. |
| method_settings (require_authorization_for_cache_control) | Specifies whether authorization is required for a cache invalidation request. |
| method_settings (unauthorized_cache_control_header_strategy) | Specifies how to handle unauthorized requests for cache invalidation. |

> | variables | A map that defines the stage variables for a Stage resource. |

Reviewer: ```suggestion
| variables | A map that defines the stage variables for a stage resource. |
```

| documentation_version | The version of the associated API documentation. |
| access_log_settings.format | A single line format of the access logs of data. |
| access_log_settings.destination_arn | The Amazon Resource Name (ARN) of the CloudWatch Logs log group or Kinesis Data Firehose delivery stream to receive access logs. |
| canary_settings.percent_traffic | The percent (0-100) of traffic diverted to a canary deployment. |
| canary_settings.deployment_id | The identifier for the canary settings deployment. |
| canary_settings.stage_variable_overrides | Stage variables overridden for a canary release deployment, including new stage variables introduced in the canary. |

> | canary_settings.use_stage_cache | A Boolean flag to indicate whether the canary deployment uses the stage cache or not. |

Reviewer: ```suggestion
| canary_settings.use_stage_cache | A boolean flag to indicate whether the canary deployment uses the stage cache or not. |
```

| tracing_enabled | Specifies whether active tracing with X-ray is enabled for the Stage. |
| web_acl_arn | The ARN of the WebAcl associated with the Stage. |
| tags | The collection of tags. Each tag element is associated with a given resource. |
| last_updated_date | The timestamp when the stage last updated. |

## Examples

### Ensure that the stage name exists.

    describe aws_api_gateway_stage(rest_api_id: 'REST_API_ID', stage_name: 'STAGE_NAME') do
      its('stage_name') { should eq 'STAGE_NAME' }
    end

### Ensure that the client certificate ID exists.

    describe aws_api_gateway_stage(rest_api_id: 'REST_API_ID', stage_name: 'STAGE_NAME') do

>         its('client_certificate_id') { should eq 'CLIENT_CETIFICATE_ID' }

Reviewer: ```suggestion
        its('client_certificate_id') { should eq 'CLIENT_CERTIFICATE_ID' }
```

Author: fixed. @IanMadd

platform: aws
---

# aws\_api\_gateway\_stage

Use the `aws_api_gateway_stage` InSpec audit resource to test properties of a single AWS ApiGateway Stage.

The AWS::ApiGateway::Stage resource deploys an API Gateway RestApi resource to a stage so that clients can call the API over the internet. The stage acts as an environment.

## Syntax

Ensure that the stage exists.

    describe aws_api_gateway_stage(rest_api_id: 'REST_API_ID', stage_name: 'STAGE_NAME') do
      it { should exist }
    end

## Parameters

`rest_api_id` _(required)_

The string identifier of the associated RestApi.

`stage_name` _(required)_

The identifier of the stage resource to get information about.

For additional information, see the [AWS ApiGateway Stage documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html).

## Properties

| Property | Description|
| --- | --- |
| deployment_id | The identifier for the deployment attached to stage resource. |
| description | The description for the stage resource. |
| created_date | The date and time that the stage resource was created. |
| stage_name | The name for the stage resource. |

> | client_certificate_id | The client certificate identifier for the stage resource. |

Reviewer 2: @Rohit1509: Can you add the following attributes in the properties section in the docs file? Like how you have added the "deployment_id" and other attributes.

**Following attributes that are missing:**  cache_cluster_enabled, cache_cluster_size, cache_cluster_status, method_settings, method_settings["String"].metrics_enabled, method_settings["String"].logging_level, method_settings["String"].data_trace_enabled, method_settings["String"].throttling_burst_limit, method_settings["String"].throttling_rate_limit, method_settings["String"].caching_enabled, method_settings["String"].cache_ttl_in_seconds, method_settings["String"].cache_data_encrypted, method_settings["String"].require_authorization_for_cache_control, method_settings["String"].unauthorized_cache_control_header_strategy"SUCCEED_WITHOUT_RESPONSE_HEADER", variables, documentation_version, access_log_settings.format, access_log_settings.destination_arn, canary_settings.percent_traffic, canary_settings.deployment_id, canary_settings.stage_variable_overrides, canary_settings.use_stage_cache, tracing_enabled, web_acl_arn, tags, last_updated_date

> | client_certificate_id | The client certificate identifier for the stage resource. |

Author: Hi @Nirbhay1997 @soumyo13 , I have added all the attributes in the doc.

> | client_certificate_id | The client certificate identifier for the stage resource. |

Reviewer 2: @Rohit1509 I checked the file and I am not able to see the changes. Have you pushed the changes?

> | client_certificate_id | The client certificate identifier for the stage resource. |

Author: @soumyo13 yes I have. My last commit with '4115988' hash contains those changes

> | client_certificate_id | The client certificate identifier for the stage resource. |

Reviewer 2: @Rohit1509 : I am able to see the changes now. Thanks.
